proto/mls/api/v1/mls.proto

```diff
@@ -78,15 +78,47 @@ service MlsApi {
       body: "*"
     };
   }
+
+  // Query stored group messages
+  rpc QueryGroupMessages(QueryGroupMessagesRequest) returns (QueryGroupMessagesResponse) {
+    option (google.api.http) = {
+      post: "/mls/v1/query-group-messages"
+      body: "*"
+    };
+  }
+
+  // Query stored group messages
+  rpc QueryWelcomeMessages(QueryWelcomeMessagesRequest) returns (QueryWelcomeMessagesResponse) {
+    option (google.api.http) = {
+      post: "/mls/v1/query-welcome-messages"
+      body: "*"
+    };
+  }
+
+  // Subscribe stream of new group messages
+  rpc SubscribeGroupMessages(SubscribeGroupMessagesRequest) returns (stream xmtp.mls.message_contents.GroupMessage) {
```

Comment:
We've previously been using `Subscribe2` for subscriptions. That requires bidirectional streaming to update the request without ending/restarting the subscription.

Reply:
What do you think about supporting subscribe from cursor instead of bidirectional streaming? IIRC that's why we added it anyway right, to mitigate the issue of missing messages during end/restart?

I'm not sure we'll have bidirectional streaming with the gateway's JSON RPC, and subscribe/continue from cursor seems like a more robust way of dealing with that issue anyway because it works for HTTP too.

Reply:
Hey, if you can implement subscribe-from-cursor across multiple topics in a reasonably performant way...I'm all for it.

Reply:
Love this! Would significantly improve reliability and help simplify client logic too